test/http.cc

```diff
@@ -50,12 +50,7 @@ TEST_P(HttpTest, Get) {
           typename decltype(e)::template ErrorsFrom<void, std::tuple<>>,
           std::tuple<std::runtime_error>>);
 
-  auto [future, k] = PromisifyForTest(std::move(e));
-  k.Start();
-
-  EventLoop::Default().RunUntil(future);
-
-  auto response = future.get();
+  auto response = *e;
```

Comment:
Another good example of where we need to `std::move(e)` here and below or better yet wrap in a lambda like you did for the filesystem tests!